src/testdir/test_vim9_class.vim

```diff
@@ -588,7 +588,37 @@ def Test_member_any_used_as_object()
   v9.CheckSourceFailure(lines, 'E1326: Member not found on object "Inner": someval')
 enddef
 
+" Nested assignment to a object variable which is of another class type
+def Test_assignment_nested_type()
+  var lines =<< trim END
+    vim9script
+
+    class Inner
+      this.value: number = 0
+    endclass
+
+    class Outer
+      this.inner: Inner
+    endclass
+
+    def F(outer: Outer)
+      outer.inner.value = 1
+    enddef
```

Comment:
Shouldn't this fail because writing  to a read-only variable outside of the class that defined it?

Reply:
Yes.  This should fail.  This is a bug.

Reply:
I have updated the PR with a fix for this issue.  Let me know if this works properly.